engine/src/main/java/org/pentaho/di/pan/EnhancedPanCommandExecutor.java

```diff
@@ -0,0 +1,528 @@
+    }
+
+    // Set metrics gathering
+    if ( "Y".equalsIgnoreCase( params.getMetrics() ) ) {
+      config.setGatheringMetrics( true );
+    }
+
+    // Apply run configuration if specified
+    if ( !Utils.isEmpty( params.getRunConfiguration() ) ) {
+      config.setRunConfiguration( params.getRunConfiguration() );
+    }
+
+    return config;
+  }
+
+  // Getter for the transformation delegate
+  public PanTransformationDelegate getTransformationDelegate() {
+    return transformationDelegate;
+  }
+
+  // Setter for the transformation delegate
+  public void setTransformationDelegate( PanTransformationDelegate transformationDelegate ) {
+    this.transformationDelegate = transformationDelegate;
+  }
+
+  /**
+   * Get the repository instance. If not already initialized, it will be null.
+   * Call initializeRepository() first to set up the repository connection.
+   */
+  public Repository getRepository() {
+    return repository;
+  }
+
+  /**
+   * Initialize repository connection based on parameters.
+   * This method extracts the repository initialization logic from PanCommandExecutor.
+   */
+  public void initializeRepository( Params params ) throws KettleException {
+
+    // Check if repository parameters are provided
+    if ( !Utils.isEmpty( params.getRepoName() ) && !isEnabled( params.getBlockRepoConns() ) ) {
+
+      /**
+       * if set, _trust_user_ needs to be considered. See pur-plugin's:
+       *
+       * @link https://github.com/pentaho/pentaho-kettle/blob/8.0.0.0-R/plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/PurRepositoryConnector.java#L97-L101
+       * @link https://github.com/pentaho/pentaho-kettle/blob/8.0.0.0-R/plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/WebServiceManager.java#L130-L133
+       */
+      if ( isEnabled( params.getTrustRepoUser() ) ) {
+        System.setProperty( "pentaho.repository.client.attemptTrust", "Y" );
+      }
+
+      // Load repository metadata
+      RepositoryMeta repositoryMeta = loadRepositoryConnection(
+        params.getRepoName(),
+        "Pan.Log.LoadingAvailableRep",
+        "Pan.Error.NoRepsDefined",
+        "Pan.Log.FindingRep"
+      );
+
+      if ( repositoryMeta == null ) {
+        getLog().logError( BaseMessages.getString( getPkgClazz(), "Pan.Error.CanNotConnectRep" ) );
+        throw new KettleException( "Could not connect to repository: " + params.getRepoName() );
+      }
+
+      // Establish repository connection
+      this.repository = establishRepositoryConnection(
+        repositoryMeta,
+        params.getRepoUsername(),
+        params.getRepoPassword(),
+        RepositoryOperation.EXECUTE_TRANSFORMATION
+      );
+    } else {
+      // No repository parameters provided, keep repository as null
+      this.repository = null;
+    }
+  }
+
+  @Override
+  public void setLog( LogChannelInterface log ) {
+    super.setLog( log );
+    if ( this.transformationDelegate == null ) {
```

Comment:
you have to find a better place to lazy load the `this.transformationDelegate `object.

you should move this to its respective getter `#getTransformationDelegate`